src/algorithms/peps_opt.jl

```diff
@@ -102,23 +170,65 @@ Evaluating the gradient of the cost function for CTMRG:
 =#
 
 function _rrule(
-    gradmode::Union{GradMode,KrylovKit.LinearSolver},
+    gradmode::GradMode{:diffgauge},
     ::RuleConfig,
     ::typeof(MPSKit.leading_boundary),
     envinit,
     state,
     alg::CTMRG,
 )
-    envs = leading_boundary(envinit, state, alg)
-    #TODO: fixed space for unit cells
+    envs = leading_boundary(envinit, state, alg)  #TODO: fixed space for unit cells
 
-    function leading_boundary_pullback(Δenvs′)
+    function leading_boundary_diffgauge_pullback(Δenvs′)
         Δenvs = unthunk(Δenvs′)
 
         # find partial gradients of gauge_fixed single CTMRG iteration
         # TODO: make this rrule_via_ad so it's zygote-agnostic
         _, env_vjp = pullback(state, envs) do A, x
-            return gauge_fix(x, ctmrg_iter(A, x, alg)[1])
+            return gauge_fix(x, ctmrg_iter(A, x, alg)[1])[1]
+        end
+
+        # evaluate the geometric sum
+        ∂f∂A(x)::typeof(state) = env_vjp(x)[1]
+        ∂f∂x(x)::typeof(envs) = env_vjp(x)[2]
+        ∂F∂envs = fpgrad(Δenvs, ∂f∂x, ∂f∂A, Δenvs, gradmode)
+
+        return NoTangent(), ZeroTangent(), ∂F∂envs, NoTangent()
+    end
+
+    return envs, leading_boundary_diffgauge_pullback
+end
+
+# Here f is differentiated from an pre-computed SVD with fixed U, S and V
+function _rrule(
+    gradmode::GradMode{:fixed},
+    ::RuleConfig,
+    ::typeof(MPSKit.leading_boundary),
+    envinit,
+    state,
+    alg::CTMRG{C},
+) where {C}
+    @assert C == :simultaneous
```

Comment:
```suggestion
    @assert C === :simultaneous
```
symbols should be slightly more efficient like this